File: SnowRoll/Assets/Scripts/Libs/AuxComponent/AuxUIComponent/AuxInputField.cs
using UnityEngine;
using UnityEngine.UI;

namespace SDK.Lib
{
    public class AuxInputField : AuxWindow
    {
        protected InputField mInputField;     // 输入

        public AuxInputField(GameObject go = null, BtnStyleID styleId = BtnStyleID.eBSID_None)
        {
            this.mSelfGo = go;
            this.updateCom();
        }

        public AuxInputField(GameObject pntNode, string path, BtnStyleID styleId = BtnStyleID.eBSID_None)
        {
            this.mSelfGo = UtilApi.TransFindChildByPObjAndPath(pntNode, path);
            this.mInputField = UtilApi.getComByP<InputField>(pntNode, path);
        }

        public void setSelfGo(GameObject pntNode, string path)
        {
            this.mSelfGo = UtilApi.TransFindChildByPObjAndPath(pntNode, path);
            this.updateCom();
        }

        public void updateCom()
        {
            if(null != this.mSelfGo)
            {
                this.mInputField = UtilApi.getComByP<InputField>(this.mSelfGo);
            }
        }

        public string text
        {
            get
            {
                return this.mInputField.text;
            }
            set
            {
                this.mInputField.text = value;
            }
        }

        public string getText()
        {
            return this.mInputField.text;
        }

        public void setText(string value)
        {
            this.mInputField.text = value;
        }

        public Color getColor()
        {
            return this.mInputField.textComponent.color;
        }

        public void setColor(Color value)
        {
            this.mInputField.textComponent.color = value;
        }
    }
}

File: Bak/scripts/MyLua/Libs/Auxiliary/AuxUIComponent/AuxInputField.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");
MLoader("MyLua.Libs.Core.GObject");
MLoader("MyLua.Libs.Auxiliary.AuxUIComponent.AuxWindow");
MLoader("MyLua.Libs.Auxiliary.AuxUIComponent.AuxUITypeId");

local M = GlobalNS.Class(GlobalNS.AuxWindow);
M.clsName = "AuxInputField";
GlobalNS[M.clsName] = M;

function M:ctor(...)
	self.mTextStr = "";
	self.mIsTextStrDirty = false;
    local pntNode, path, styleId = ...;
	
    if(styleId == nil) then
        styleId = GlobalNS.BtnStyleID.eBSID_None;
    end
	
    self.mSelfGo = GlobalNS.UtilApi.TransFindChildByPObjAndPath(pntNode, path);
    self.mInputField = GlobalNS.UtilApi.getComByPath(pntNode, path, GlobalNS.AuxUITypeId.InputField);
end

function M:onSelfChanged()
	M.super.onSelfChanged(self);
	
	self:syncText();
end

function M:setText(value)
	if(self.mTextStr ~= value) then
		self.mIsTextStrDirty = true;
		self.mTextStr = value;
	end
	
	self:syncText();
end

function M:getText()
    --return self.mInputField.text;
	return self.mTextStr;
end

function M:syncText()
	if(nil ~= self.mInputField) then
		if(self.mIsTextStrDirty) then
			self.mIsTextStrDirty = false;
			
			if(self.mInputField.text ~= self.mTextStr) then
				self.mInputField.text = self.mTextStr;
			end
		else
			self.mTextStr = self.mInputField.text;
		end
	end
end

return M;